front/lib/models.ts

```diff
@@ -1241,6 +1241,187 @@ DataSource.hasMany(DocumentTrackerChangeSuggestion, {
   onDelete: "CASCADE",
 });
 
+// Assistant V2
+export class Conversation extends Model<
+  InferAttributes<Conversation>,
+  InferCreationAttributes<Conversation>
+> {
+  declare id: number;
+  declare title: string;
+  declare created: Date;
+  declare visibility: string;
+}
+
+Conversation.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    title: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    created: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    visibility: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "conversation",
+    sequelize: front_sequelize,
+  }
+);
+
+export class UserMessage extends Model<
+  InferAttributes<UserMessage>,
+  InferCreationAttributes<UserMessage>
+> {
+  declare id: number;
+  declare textContent: string;
+  declare userId: ForeignKey<User["id"]>;
+}
+
+UserMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    textContent: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "user_message",
+    sequelize: front_sequelize,
+  }
+);
+
+UserMessage.belongsTo(User, { foreignKey: "userId" });
+User.hasMany(UserMessage, { foreignKey: "userId" });
+
+export class AssistantMessage extends Model<
+  InferAttributes<AssistantMessage>,
+  InferCreationAttributes<AssistantMessage>
+> {
+  declare id: number;
+  declare textContent: string | null;
+}
+
+AssistantMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    textContent: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "assistant_message",
+    sequelize: front_sequelize,
+  }
+);
+
+export class Message extends Model<
+  InferAttributes<Message>,
+  InferCreationAttributes<Message>
+> {
+  declare id: number;
+
+  declare conversation_id: number;
+  declare version: number;
+  declare rank: number;
+
+  declare is_deleted: boolean;
```

Comment:
@spolu  Is there a strong reason to prefer `status: "visible" | "deleted"` over a boolean ?

Reply:
My 5 years at Stripe has taught me on thing: always status never a boolean